### PR TITLE
Init Check for the config and typo corrections

### PR DESCRIPTION
**Pull request by BShrivastav**
- Added redundancy/validations check for inputs:
     - region name
     - custom image name
     - resource_group_id
     - security_group_id
- Removed the crash.log
- Typo correction with README.md

### CODE REVIEW

File: builder/ibmcloud/vpc/step_get_subnet_info.go [hunk 1]
```diff
@@ -34,6 +34,14 @@ func (s *stepGetSubnetInfo) Run(_ context.Context, state multistep.StateBag) mul
 	}
 
 	vpcId := *subnetData.VPC.ID
+	secGrpVPC := state.Get("user_sec_grp_vpc")
+	ui.Say("Verifying the security group and subnet belongs to same VPC..")
+	if vpcId != secGrpVPC {
+		err := fmt.Errorf("[ERROR] Error VPCs of provided security group %s, and subnet's VPC %s, doesnot match", secGrpVPC, vpcId)
```
Comment (astha-jain) [inline builder/ibmcloud/vpc/step_get_subnet_info.go:40]: Please use the security group and subnet connected to VPC id %s

File: builder/ibmcloud/vpc/step_verify_input.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+		Name: &config.Region,
+	}
+	_, _, err := vpcService.GetRegion(getRegionOptions)
+	if err != nil {
+		err := fmt.Errorf("[ERROR] Error fetching region : %s: %s", config.Region, err)
+		state.Put("error", err)
+		ui.Error(err.Error())
+		return multistep.ActionHalt
+	}
+	ui.Say("Region verified, check Passed")
```
Comment (astha-jain) [inline builder/ibmcloud/vpc/step_verify_input.go:37]: need to rephrase this

File: builder/ibmcloud/vpc/step_verify_input.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+		serviceClientOptions := &resourcemanagerv2.ResourceManagerV2Options{
+			Authenticator: &core.IamAuthenticator{
+				ApiKey: client.IBMApiKey,
+				URL:    config.IAMEndpoint,
+			},
+		}
+		serviceClient, err := resourcemanagerv2.NewResourceManagerV2UsingExternalConfig(serviceClientOptions)
+		if err != nil {
+			err := fmt.Errorf("[ERROR] Error fetching resource group : %s: %s", config.ResourceGroupID, err)
```
Comment (astha-jain) [inline builder/ibmcloud/vpc/step_verify_input.go:50]: Incorrect error message.

File: builder/ibmcloud/vpc/step_verify_input.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+			state.Put("error", err)
+			ui.Error(err.Error())
+			return multistep.ActionHalt
+		} else if result == nil {
+			err := fmt.Errorf("[ERROR] Resource group not found resource_group_id : %s: %s", config.ResourceGroupID, err)
+			state.Put("error", err)
+			ui.Error(err.Error())
+			return multistep.ActionHalt
+		} else {
+			ui.Say(fmt.Sprintf("Continuing with resource group : %s", *result.Name))
```
Comment (astha-jain) [inline builder/ibmcloud/vpc/step_verify_input.go:67]: Do we need this?

File: builder/ibmcloud/vpc/step_verify_input.go [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+		}
+	}
+	// image check
+	ui.Say(fmt.Sprintf("Checking the custom image: %s for redundancy", config.ImageName))
```
Comment (astha-jain) [inline builder/ibmcloud/vpc/step_verify_input.go:71]: same, do we need this to be displayed?

File: builder/ibmcloud/vpc/step_verify_input.go [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+	if err != nil {
+		err := fmt.Errorf("[ERROR] Error checking custom image %s", err)
+		state.Put("error", err)
+		ui.Error(err.Error())
+		return multistep.ActionHalt
+	}
+	allrecs := availableImages.Images
+
+	if len(allrecs) != 0 {
+		err := fmt.Errorf("[ERROR] Existing custom image found with name: %s", config.ImageName)
```
Comment (astha-jain) [inline builder/ibmcloud/vpc/step_verify_input.go:90]: Maybe "Could not find an image with the name %s"

File: builder/ibmcloud/vpc/step_verify_input.go [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+		}
+		secGrp, _, err := vpcService.GetSecurityGroup(secgrpOption)
+		if err != nil {
+			err := fmt.Errorf("[ERROR] Error fetching security group %s", err)
+			state.Put("error", err)
+			ui.Error(err.Error())
+			return multistep.ActionHalt
+		}
+		if *secGrp.ID != "" {
+			ui.Say(fmt.Sprintf("Using provided security group: %s with security_group_id: %s", *secGrp.Name, *secGrp.ID))
```
Comment (astha-jain) [inline builder/ibmcloud/vpc/step_verify_input.go:111]: Again, do we need this?

**Review by deepaksibm — APPROVED** [review]

File: builder/ibmcloud/vpc/step_get_subnet_info.go [hunk 8]
```diff
@@ -34,6 +34,14 @@ func (s *stepGetSubnetInfo) Run(_ context.Context, state multistep.StateBag) mul
 	}
 
 	vpcId := *subnetData.VPC.ID
+	secGrpVPC := state.Get("user_sec_grp_vpc")
```
Comment (ujjwal-ibm) [inline builder/ibmcloud/vpc/step_get_subnet_info.go:37]: This is throwing an error saying security group is not in same vpc. Even when not passing sg id

**Review by ujjwal-ibm — CHANGES_REQUESTED** [review]

**Review by astha-jain — APPROVED** [review]

**Pull request merged**